File: frontend/src/types/trip.ts
export interface TripRequest {
  destination: string;
  duration: number;
  budget: number;
  travelers: number;
  preferences: string[];
}

export interface Trip {
  id: string;
  destination: string;
  duration: number;
  budget: number;
  travelers: number;
  preferences: string[];
  plan?: string;
  status: 'pending' | 'processing' | 'completed' | 'failed';
}

File: backend/app/schemas/trip.py
from pydantic import BaseModel
from typing import Optional, List


class TripRequest(BaseModel):
    destination: str
    duration: int
    budget: float
    travelers: int
    preferences: List[str]


class TripResponse(BaseModel):
    id: str
    destination: str
    duration: int
    budget: float
    travelers: int
    preferences: List[str]
    plan: Optional[str] = None
    status: str = "pending"


class TripOptimizeRequest(BaseModel):
    optimization_type: str  # "budget", "time", "experience"
